app/controllers/fhir/r4/api_controller.rb

```diff
@@ -0,0 +1,374 @@
+  #
+  # GET /[:resource_type]/[:id]
+  def show
+    resource_type = params.permit(:resource_type)[:resource_type]&.downcase
+    case resource_type
+    when 'patient'
+      resource = get_patient(params.permit(:id)[:id])
+    when 'observation'
+      resource = get_laboratory(params.permit(:id)[:id])
+    when 'questionnaireresponse'
+      resource = get_assessment(params.permit(:id)[:id])
+    else
+      status_bad_request && return
+    end
+
+    status_forbidden && return if resource.nil?
+
+    status_ok(resource.as_fhir) && return
+  rescue StandardError
+    render json: operation_outcome_fatal.to_json, status: :internal_server_error
+  end
+
+  # Update a resource given a type and an id.
+  #
+  # Supports (updating): Patient
+  #
+  # PUT /fhir/r4/[:resource_type]/[:id]
+  def update
+    # Parse in the FHIR::Patient
+    contents = FHIR.from_contents(request.body.string)
+    status_bad_request && return if contents.nil? || !contents.valid?
+
+    resource_type = params.permit(:resource_type)[:resource_type]&.downcase
+    case resource_type
+    when 'patient'
+      updates = Patient.from_fhir(contents).select { |_k, v| v.present? }
+      resource = get_patient(params.permit(:id)[:id])
+    else
+      status_bad_request && return
+    end
+
+    status_forbidden && return if resource.nil?
+
+    # Try to update the resource
+    status_bad_request && return if updates.nil? || !resource.update(updates)
+
+    status_ok(resource.as_fhir) && return
+  rescue StandardError
+    render json: operation_outcome_fatal.to_json, status: :internal_server_error
+  end
+
+  # Create a resource given a type.
+  #
+  # Supports (writing): Patient
+  #
+  # POST /fhir/r4/[:resource_type]
+  def create
+    # Parse in the FHIR::Patient
+    contents = FHIR.from_contents(request.body.string)
+    status_bad_request && return if contents.nil? || !contents.valid?
+
+    resource_type = params.permit(:resource_type)[:resource_type]&.downcase
+    case resource_type
+    when 'patient'
+      # Construct a Sara Alert Patient
+      resource = Patient.new(Patient.from_fhir(contents))
+
+      # Responder is self
+      resource.responder = resource
+
+      # Creator is authenticated user
+      resource.creator = current_resource_owner
+
+      # Jurisdiction is the authenticated user's jurisdiction
+      resource.jurisdiction = current_resource_owner.jurisdiction
+
+      # Generate a submission token for the new monitoree
+      resource.submission_token = SecureRandom.hex(20) # 160 bits
+    end
+
+    status_bad_request && return if resource.nil?
+
+    status_bad_request && return unless resource.save
+
+    if resource_type == 'patient'
+      # Send enrollment notification
+      resource.send_enrollment_notification
+
+      # Create a history for the enrollment
+      history = History.new
+      history.created_by = current_resource_owner.email
+      history.comment = 'User enrolled monitoree.'
```

Comment:
not that important, but would it be worth noting that the user was enrolled via the api? (same for imports now that I think of it), just an idea/suggestion, but could be something along the lines of:

normal enrollment process: User enrolled monitoree
import: User enrolled monitoree via import
api: User enrolled monitoree via api